internal/chezmoi/findexecutable.go

```diff
@@ -0,0 +1,47 @@
+package chezmoi
+
+import (
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+var (
+	foundExecutableCacheMutex sync.Mutex
+	foundExecutableCache      = make(map[string]struct{})
+)
+
+// FindExecutable is like lookPath except that you can specify the paths rather than just using the current `$PATH`. This
+// makes it useful for the resulting path of rc/profile files.
+func FindExecutable(file string, paths ...string) (string, error) {
+	foundExecutableCacheMutex.Lock()
+	defer foundExecutableCacheMutex.Unlock()
+
+	// stolen from: /usr/lib/go-1.20/src/os/exec/lp_unix.go:52
+	for _, dir := range paths {
```

Comment:
This is inefficient because it recomputes `findExecutableExtensions` every time.

Add:

```go
candidates := findExecutableExtensions(file)
```

outside of this loop.

Reply:
You are correct since the path doesn't matter. Will address on update